daemon/src/model/cfd.rs

```diff
@@ -805,6 +851,51 @@ impl Cfd {
                     self.state
                 ),
             },
+            CfdStateChangeEvent::ProposalSigned(transaction) => match self.state.clone() {
+                CfdState::Open {
+                    common,
+                    dlc,
+                    attestation,
+                    ..
+                } => CfdState::Open {
+                    common,
+                    dlc,
+                    attestation,
+                    collaborative_close: Some((transaction, SystemTime::now())),
+                },
+                CfdState::PendingCommit {
```

Comment:
That is an interesting one - this transition is also mostly for completion, but in a real world scenario is very weird :D
Something along the lines of "we proposed collaborative settlement but then hit commit anyway before the maker came back to us to actually confirm the collaborative settlement". I think such a scenario will result in the maker  failing to send the collaboratively constructed transaction and we will have to use the cet.

Reply:
This has been overtaken by changes, `PendingCommit` does not have `collaborative_settlement` anymore :)